### PR TITLE
[SPEC-260] Add analytics to the installer and setup telemetry

### PR DESCRIPTION
**Pull request by peterjgrainger**
- Created a dotnet core cli targeting .Net 4.5 (same as speckle core)
- Accepts three arguments `version` `domain` e.g. `1.0.10.32423` `my-domain`
- Checks if the file SpeckleUpdater.exe exists. If so, this is an update not a new install
- Call this CLI after installation and only for the supplied domain on installer creation time (allowed domain is stored in secrets)
- Update the SQLite database and set telemetry to true so clients know if to send telemetry 

@daviddekoning @jenessaman please check my c#. It's my first .net core program :)

Can view the data like this
image

### CODE REVIEW

File: Analytics/Program.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+        static void Main(string[] args)
+        {
+            PiwikTracker _piwikTracker = new PiwikTracker(SiteId, PiwikBaseUrl);
+            String _version = args[0];
+            String _allowedDomain = args[1];
+            String _machineName = Environment.MachineName.ToLower(new CultureInfo("en-GB", false));            
+            String _appDataFolder = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+            String _speckleUpdaterExe = _appDataFolder + "\\Speckle\\SpeckleUpdater.exe";
+
```
Comment (daviddekoning) [inline Analytics/Program.cs:24]: let's set the userId. Given that we are only tracking Arup employees, we should be able to generate their email as 'windows username' + '@arup.com'

File: Analytics/Program.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            // Check to see if this is an update or it's a new user.
+            String _installType = File.Exists(_speckleUpdaterExe) ? "update" : "new";
+            
+            // Only track users from specific domains (not general public)
+            if(_machineName.Contains(_allowedDomain)) {
+                // Sets a flag in the cache database. Used by clients.
+                LocalContext.SetTelemetrySettings(true);
+                // Send this information to Matomo
+                _piwikTracker.DoTrackEvent("SpeckleInstaller", _installType, _version); 
+            }
```
Comment (daviddekoning) [inline Analytics/Program.cs:34]: LocalContext sets Telemetry to true by default, so we should explicitly set it to false if it is not an allowed domain

File: Analytics/Program.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+using SpeckleCore;
+using System.Globalization;
+using System.Diagnostics;
+using System.IO;
+
+namespace analytics
+{
+    class Program
+    {
+        private static readonly string PiwikBaseUrl = "https://arupdt.matomo.cloud/";
```
Comment (daviddekoning) [inline Analytics/Program.cs:13]: are there any security concerns about leaving this in the code?

**Review by daviddekoning — CHANGES_REQUESTED** [review]
thanks, Peter. Overall, it looks good to me - I've noted a couple of things to update.

File: Analytics/Program.cs (hunk 3 above)
Comment (peterjgrainger) [reply Analytics/Program.cs:11]: It's not a secret. Did a search in GitHub and got 34k results for `matomo.cloud` https://github.com/search?q=matomo.cloud%2F&type=code

File: Analytics/Program.cs (hunk 2 above)
Comment (peterjgrainger) [reply Analytics/Program.cs:32]: Good idea.

File: Analytics/Program.cs (hunk 1 above)
Comment (peterjgrainger) [reply Analytics/Program.cs:22]: Agreed.

**Comment by daviddekoning** [comment]
Thanks, @peterjgrainger - this all looks good to me

**Pull request merged**